src/components/SignInButton.tsx

```diff
@@ -0,0 +1,212 @@
+import React, { useState } from 'react';
+import { isEmpty, isNil } from 'lodash/fp';
+import { Alert } from './Alert';
+import { ToS } from '../libs/ajax/ToS';
+import { DuosUser, User } from '../libs/ajax/User';
+import { Metrics } from '../libs/ajax/Metrics';
+import { Storage } from '../libs/storage';
+import { Navigation, setUserRoleStatuses } from '../libs/utils';
+import loadingImage from '../images/loading-indicator.svg';
+import { Auth } from '../libs/auth/auth';
+import eventList from '../libs/events';
+import { StackdriverReporter } from '../libs/stackdriverReporter';
+import CSS from 'csstype';
+import { useHistory } from 'react-router';
+import { OidcUser } from 'src/libs/auth/oidcBroker';
+
+interface SignInButtonProps {
+  customStyle: CSS.Properties | undefined;
+}
+
+interface ErrorInfo {
+  title?: string;
+  description?: string;
+}
+
+interface HttpError extends Error {
+  status?: number;
+}
+
+export const SignInButton = (props: SignInButtonProps) => {
+  const [errorInfo, setErrorInfo] = useState<ErrorInfo>({});
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [showError, setShowError] = useState<boolean>(false);
+  const { customStyle } = props;
+  const history = useHistory();
+
+  // Utility function called in the normal success case and in the undocumented 409 case
```

Comment:
We should look more into this - I think with the update to v2 ToS APIs, the 409 case is no longer a thing.